tests/v2_validation/cattlevalidationtest/core/test_cattle_upgrade.py

```diff
@@ -0,0 +1,291 @@
+                    preportsuffixnum, socat_containers)
+
+
+def create_stacks(admin_client, client, stack_name1, stack_name2,
+                  portsuffixnum):
+
+    # Create pre-upgrade stack
+    print "**** In Create Stacks ****"
+
+    print "PORT SUFFIX NUM"
+
+    print portsuffixnum
+    dc_config_file1 = "dc_v1.3.3.yml"
+    rc_config_file1 = "rc_v1.3.3.yml"
+    dc_config1 = readDataFile(UPGRADE_SUBDIR, dc_config_file1)
+    dc_config1 = dc_config1.replace("$portsuffixnum", portsuffixnum)
+
+    print dc_config1
+
+    rc_config1 = readDataFile(UPGRADE_SUBDIR, rc_config_file1)
+    rc_config1 = rc_config1.replace("$portsuffixnum", portsuffixnum)
+
+    print rc_config1
+    create_stack_with_service_from_config(client, stack_name1, dc_config1,
+                                          rc_config1)
+
+    dc_config_file2 = "dc_newst_v1.3.3.yml"
+    rc_config_file2 = "rc_newst_v1.3.3.yml"
+    dc_config2 = readDataFile(UPGRADE_SUBDIR, dc_config_file2)
+    dc_config2 = dc_config2.replace("$stack", stack_name1)
+
+    rc_config2 = readDataFile(UPGRADE_SUBDIR, rc_config_file2)
+    rc_config2 = rc_config2.replace("$stack", stack_name1)
+
+    create_stack_with_service_from_config(client, stack_name2, dc_config2,
+                                          rc_config2)
+
+
+def validate_stacks(admin_client, client, stackname,
+                    portsuffixnum, socat_containers):
+
+    stack1 = stackname + "-1"
+    stack2 = stackname + "-2"
+    print "In validate stacks"
+
+    # Validate the containers/lbs in the stack
+
+    stack, service1 = get_env_service_by_name(client, stack1, "service1")
+    assert service1['state'] == "active"
+    assert service1.scale == 2
+
+    # Validate LB Service
+
+    stack, lbservice = get_env_service_by_name(client, stack1, "mylb")
+
+    assert lbservice['state'] == "active"
+    assert lbservice.scale == 1
+
+    mylbport = "300" + str(portsuffixnum)
+    validate_lb_service(admin_client, client, lbservice,
+                        mylbport, [service1])
+    # Validate Health LB Service
+
+    stack, healthservice = get_env_service_by_name(client, stack1,
+                                                   "healthservice")
+    assert service1['state'] == "active"
+    assert healthservice.scale == 1
+
+    stack, healthlbservice = get_env_service_by_name(client, stack1,
+                                                     "healthlb")
+    assert healthlbservice['state'] == "active"
+    assert healthlbservice.scale == 1
+
+    healthlbport = "200" + str(portsuffixnum)
+    validate_lb_service(admin_client, client, healthlbservice,
+                        healthlbport, [healthservice])
+
+    # Validate Global Health LB Service
+    stack, globalhealthservice = get_env_service_by_name(client, stack1,
+                                                         "globalhealthservice")
+    assert globalhealthservice['state'] == "active"
```

Comment:
check that the scale of service is equal to number of host in environment.
Check that there is 1 instance/host.